utils.go

```diff
@@ -2,17 +2,57 @@ package main
 
 import (
 	"context"
+	"net/http"
 
+	"github.com/gin-gonic/gin"
+	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
 )
 
 var (
 	httpAuthorizationHeader = "Authorization"
 	grpcAuthorizationHeader = "authorization"
 )
 
+var grpcCodeToHttpCode = map[codes.Code]int{
+	codes.OK:                 http.StatusOK,
+	codes.Canceled:           http.StatusInternalServerError,
+	codes.Unknown:            http.StatusInternalServerError,
+	codes.InvalidArgument:    http.StatusBadRequest,
+	codes.DeadlineExceeded:   http.StatusRequestTimeout,
+	codes.NotFound:           http.StatusNotFound,
+	codes.AlreadyExists:      http.StatusConflict,
+	codes.PermissionDenied:   http.StatusUnauthorized,
+	codes.ResourceExhausted:  http.StatusServiceUnavailable,
+	codes.FailedPrecondition: http.StatusFailedDependency,
+	codes.Aborted:            http.StatusServiceUnavailable,
+	codes.OutOfRange:         http.StatusRequestedRangeNotSatisfiable,
+	codes.Unimplemented:      http.StatusNotImplemented,
+	codes.Internal:           http.StatusInternalServerError,
+	codes.Unavailable:        http.StatusServiceUnavailable,
+	codes.DataLoss:           http.StatusInternalServerError,
+	codes.Unauthenticated:    http.StatusUnauthorized,
+}
```

Comment:
Feedback appreciated here. I've made the mapping myself so there are surely ways we can improve this.